web/modules/custom/rdf_entity/src/Entity/Query/Sparql/SparqlCondition.php

```diff
@@ -4,47 +4,785 @@
+    }
+    return 'VALUES ' . $this->toVar($condition['field']) . ' {' . implode(' ', $value) . '}';
+  }
+
+  /**
+   * Compiles a filter condition.
+   *
+   * @param array $filter_fragments
+   *   An array of filter strings.
+   *
+   * @return string
+   *   A condition fragment string.
+   */
+  protected function compileFilters(array $filter_fragments) {
+    // The delimiter is always a '&&' because otherwise it would be a separate
+    // condition class.
+    $delimiter = '&&';
+    if (count($filter_fragments) > 1) {
+      $compiled_filter = '(' . implode(') ' . $delimiter . '(', $filter_fragments) . ')';
+    }
+    else {
+      $compiled_filter = reset($filter_fragments);
+    }
+
+    return 'FILTER (' . $compiled_filter . ')';
+  }
+
+  /**
+   * Calculates the langcode of the field if one exists.
+   *
+   * @param string $field
+   *   The field name.
+   * @param string $column
+   *   The column name.
+   * @param string $default_lang
+   *   A default langcode to return.
+   *
+   * @return string
+   *   The langcode to be used.
+   */
+  protected function getLangCode($field, $column = NULL, $default_lang = NULL) {
+    $format = $this->fieldHandler->getFieldFormat($this->query->getEntityTypeId(), $field, $column);
+    $format = reset($format);
+    if ($format !== RdfFieldHandler::TRANSLATABLE_LITERAL) {
+      return FALSE;
+    }
+
+    $non_languages = [
+      LanguageInterface::LANGCODE_NOT_SPECIFIED,
+      LanguageInterface::LANGCODE_DEFAULT,
+      LanguageInterface::LANGCODE_NOT_APPLICABLE,
+      LanguageInterface::LANGCODE_SITE_DEFAULT,
+      LanguageInterface::LANGCODE_SYSTEM,
+    ];
+
+    if (empty($default_lang) || in_array($default_lang, $non_languages)) {
+      return \Drupal::languageManager()->getCurrentLanguage()->getId();
     }
 
+    return $default_lang;
   }
 
   /**
    * Implements \Drupal\Core\Entity\Query\ConditionInterface::exists().
    */
-  public function exists($field, $langcode = NULL) {
-    $this->query->condition('_field_exists', $field, 'EXISTS');
+  public function exists($field, $lang = NULL) {
+    return $this->condition($field, NULL, 'EXISTS');
   }
 
   /**
    * Implements \Drupal\Core\Entity\Query\ConditionInterface::notExists().
    */
-  public function notExists($field, $langcode = NULL) {
-    $this->query->condition('_field_exists', $field, 'NOT EXISTS');
+  public function notExists($field, $lang = NULL) {
+    $this->condition($field, NULL, 'NOT EXISTS');
+  }
+
+  /**
+   * Prefixes a keyword with a prefix in order to be treated as a variable.
+   *
+   * @param string $key
+   *   The name of the variable.
+   * @param bool $blank
+   *   Whether or not to be a blank note.
+   *
+   * @return string
+   *   The variable.
+   */
+  protected function toVar($key, $blank = FALSE) {
+    // Deal with field.property as dots are not allowed in var names.
+    $key = str_replace('.', '_', $key);
+    if (strpos($key, '?') === FALSE && strpos($key, '_:') === FALSE) {
```

Comment:
I will keep the method as it is because we will need it in the future when we implement the blank nodes. Instead, I will remove the instances where I had a variable be set as a blank node (in the compileExists).